test/Microsoft.ML.Tests/Transformers/ValueMappingTests.cs

```diff
@@ -76,6 +77,39 @@ public void ValueMapOneValueTest()
             Assert.Equal(1, fValue);
         }
 
+        [Fact]
+        public void ValueMapInputIsVectorTest()
+        {
+            var data = new[] { new TestClass() { A = "bar test foo", B = "test", C = "foo" } };
+            var dataView = ComponentCreation.CreateDataView(Env, data);
```

Comment:
What happens if there is a string that is not in the dictionary?

Reply:
Look at the `ValueMappingMissingKey` test below. Let me know if that's what you were asking?

---
In reply to: [249197953](https://github.com/dotnet/machinelearning/pull/2190#discussion_r249197953) [](ancestors = 249197953)